tests/core/custom_types/test_coin.py

```diff
@@ -76,34 +76,34 @@ def test_construction() -> None:
     H1 = b"a" * 32
     H2 = b"b" * 32
 
-    with pytest.raises(OverflowError, match="int too big to convert"):
+    with pytest.raises(ValueError, match="Value 18446744073709551616 does not fit into uint64"):
         # overflow
-        Coin(H1, H2, 0x10000000000000000)
+        Coin(H1, H2, uint64(0x10000000000000000))
 
-    with pytest.raises(OverflowError, match="can't convert negative int to unsigned"):
+    with pytest.raises(ValueError, match="Value -1 does not fit into uint64"):
         # overflow
-        Coin(H1, H2, -1)
+        Coin(H1, H2, uint64(-1))
 
     H1_short = b"a" * 31
     H1_long = b"a" * 33
 
     with pytest.raises(ValueError):
         # short hash
-        Coin(H1_short, H2, 1)
+        Coin(H1_short, H2, uint64(1))
```

Comment:
With the super vague exception here I'm a bit uncomfortable with the `uint64` construction inside the test.  Could we have a `uint64_one` or something here constructed outside?  Where yes, the real fix is to assert about the message or even have more unique exception types (clearly not for this pr).

Or, tell me to shut up because it is `uint64(1)` which won't fail and close the conversation.

Reply:
you mean the risk is that the test doesn't detect *what* failed?

Reply:
I'll add `match=` arguments